Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -1969,21 +1969,147 @@ void WebsiteDataStore::resetStoragePersistedState(CompletionHandler<void()>&& co
 }
 
 #if !PLATFORM(COCOA)
-String WebsiteDataStore::defaultMediaCacheDirectory(const String&)
+// In https://bugs.webkit.org/show_bug.cgi?id=244058#c3 Sihui says:
+//
+//   CacheStorage is not "cache", as its lifetime should be controlled by web clients
+//   (https://w3c.github.io/ServiceWorker/#cache-lifetimes), like IndexedDB.
+//
+// But we cannot change this function to use baseDataDirectory without leaking previous storage.
+// This problem will go away when using UnifiedOriginStorageLevel::Standard, as then data will be
+// migrated from this location to the GeneralStorageDirectory.
+String WebsiteDataStore::defaultCacheStorageDirectory(const String& baseCacheDirectory)
```

Comment:
The comment should probably be put inside function as it explains why we use `baseCacheDirectory`. I prefer the comment to be more succinct like:

> This should be baseDataDirectory per spec: https://w3c.github.io/ServiceWorker/#cache-lifetimes.
> Keep using baseCacheDirectory now as we plan to use GeneralStorageDirectory for CacheStorage soon.

(I think we don't normally comment "we do this because someone said so", but "we do this because of some good reason" : P)

Reply:
OK, I reworded this to:

>     // CacheStorage is really data, not cache, as its lifetime should be controlled by web clients.
>     // https://w3c.github.io/ServiceWorker/#cache-lifetimes
>     //
>     // Keep using baseCacheDirectory for now for compatibility, to avoid leaking existing storage.
>     // Soon, it will be migrated to GeneralStorageDirectory.